rpcserver.go

```diff
@@ -344,6 +345,36 @@ func handleAddNode(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (in
 	return nil, nil
 }
 
+// handleNode handles node commands.
+func handleNode(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*btcjson.NodeCmd)
+
+	addr := normalizeAddress(c.Addr, activeNetParams.DefaultPort)
+	var err error
+	switch c.SubCmd {
+	case "disconnect":
+		err = s.server.DisconnectNode(addr)
+	case "remove":
+		err = s.server.RemoveNode(addr)
+	case "connect":
+		switch *c.ConnectSubCmd {
+		case "perm", "non-perm":
+			err = s.server.ConnectNode(addr, *c.ConnectSubCmd == "perm")
+		default:
+			err = errors.New("invalid subcommand for node connect")
```

Comment:
Noted, I've changed this fragment to return the proper error, and additionally altered `addnode` to follow the same behavior for returned errors.